class/world.lua

```diff
@@ -52,6 +54,18 @@ World = Class {
         self.grid:calculatePaths()
         return true --an obstacle was placed
     end;
+    removeStructure = function(self, structure)
+        if structure ~= nil then 
```

Comment:
this can just be `if structure then`.

Alternatively the whole if statement can be replaced with an `assert(structure)` statement which will crash the game if its not true. Sounds spicy but it should never happen, so it makes it obvious to us if something is wrong.